pkg/apis/ibmcloudprovider/v1/ibmcloudproviderconfig_types.go

```diff
@@ -36,6 +36,12 @@ type IBMCloudMachineProviderSpec struct {
 	// VPC name where the instance will be created
 	VPC string `json:"vpc"`
 
+	// BootVolume is the machine's boot volume details.
+	// When omitted, will use IBM Cloud's default Instance boot volume configuration.
+	// Such as using IBM-managed encryption for the volume.
+	// +optional
+	BootVolume *IBMCloudMachineBootVolume `json:"bootVolume,omitempty"`
```

Comment:
I think we want a pointer, as we would want to allow it not being set, in order to rely on the default boot volume the IBM Cloud VPC provider will generate for us (not having to provide those details to the VPC API). Where, when it is specified, then we will generate a boot volume template to pass in to the VPC API.

So I guess it makes sense to me to have this be a pointer, that way when converting between JSON (YAML) for the MachineProviderSpec, the `bootVolume` is nil  and we rely on IBM Cloud VPC API to default the boot volume specs, or it contains details about the boot volume we generate via a template and pass as part of the machine details to the VPC API.

But I could be using that idea incorrectly here.

Reply:
Currently at least, there is one field in here that I'm guessing should be required right? Where you check this field, you first check it's not nil, and then check the value of the field within is non-empty. If you moved this away from being a pointer, the change there would be just to get rid of the nil check. Seems it would work well and remove the pointer deref making the code simpler. Not a blocker, just an observation that I think this could simplify the code a little

Reply:
Sure, I can change it from a pointer. I'll do that quick.

Reply:
I updated to remove the pointer.

I was wondering if there would be an compatibility issues if we did move to using a pointer, or making fields optional, as although I don't know how soon, we hope to eventually support a boot volume profile, to help with etcd efficiency. I'm guessing on a minor release boundary, so not sure how/if that affects machine configs during an upgrade process (4.15 -> 4.16+).

Not sure if that would affect the expectations here or not, as I haven't thought out the design too long term TBH.

Reply:
We can change back later if need be, but for now, this seems to be the simpler way